Content.Server/SimpleStation14/Species/Shadowkin/Systems/ShadowkinSystem.cs

```diff
@@ -0,0 +1,158 @@
+                        ("power", _powerSystem.GetLevelInt(component.PowerLevel)),
+                        ("powerMax", component.PowerLevelMax),
+                        ("powerType", powerType)
+                    ));
+                }
+                else
+                {
+                    args.PushMarkup(Loc.GetString("shadowkin-power-examined-other",
+                        ("target", Identity.Entity(uid, _entityManager)),
+                        ("powerType", powerType)
+                    ));
+                }
+            }
+        }
+
+        private void OnInit(EntityUid uid, ShadowkinComponent component, ComponentInit args)
+        {
+            if (component.PowerLevel <= ShadowkinComponent.PowerThresholds[ShadowkinPowerThreshold.Min] + 1f)
+                _powerSystem.SetPowerLevel(component.Owner, ShadowkinComponent.PowerThresholds[ShadowkinPowerThreshold.Okay]);
+
+            component.ForceSwapRate = _random.NextFloat(component.ForceSwapRateMin, component.ForceSwapRateMax);
+            component.TiredRate = _random.NextFloat(component.TiredRateMin, component.TiredRateMax);
+        }
+
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            var query = _entityManager.EntityQuery<ShadowkinComponent>(true);
+
+            // Update power level for all shadowkin
+            foreach (var component in query)
+            {
+                // These MUST be  TryUpdatePowerLevel  THEN  TryBlackeye  or else init will always blackeye
+                _powerSystem.TryUpdatePowerLevel(component.Owner, frameTime);
+                if (!component.Blackeye)
+                    _powerSystem.TryBlackeye(component.Owner);
+
+
+                // Sync client and server
+                component.DirtyAccumulator += frameTime;
+                if (component.DirtyAccumulator > component.DirtyAccumulatorRate)
+                {
+                    component.DirtyAccumulator = 0f;
+
+                    _powerSystem.UpdateAlert(component.Owner, true, component.PowerLevel);
+                    Dirty(component);
+                }
+
+
+                #region Random Stuff
+
+                #region ForceSwap
+                // Check if they're at max power
+                if (component.PowerLevel >= ShadowkinComponent.PowerThresholds[ShadowkinPowerThreshold.Max])
+                {
+                    // If so, start the timer
+                    component.ForceSwapAccumulator += frameTime;
+
+                    // If the timer is up, force swap
+                    if (component.ForceSwapAccumulator > component.ForceSwapRate)
+                    {
+                        // Reset timer
+                        component.ForceSwapAccumulator = 0f;
+                        // Random new timer
+                        component.ForceSwapRate = _random.NextFloat(component.ForceSwapRateMin, component.ForceSwapRateMax);
```

Comment:
```suggestion
                    // If so, start the timer
                    component.ForceSwapAccumulator -= frameTime;

                    // If the timer is up, force swap
                    if (component.ForceSwapAccumulator =< 0)
                    {
                        // Reset timer
                        component.ForceSwapAccumulator += _random.NextFloat(component.ForceSwapRateMin, component.ForceSwapRateMax);
```
Can remove a variable :)